cockatrice/src/pictureloader.cpp

```diff
@@ -43,40 +43,80 @@ class PictureToLoad::SetDownloadPriorityComparator
     }
 };
 
-PictureToLoad::PictureToLoad(CardInfoPtr _card) : card(std::move(_card)), setIndex(0)
+PictureToLoad::PictureToLoad(CardInfoPtr _card) : card(std::move(_card))
 {
+    // This will be replaced with an expandable list ideally
+    urlTemplates.append(settingsCache->getPicUrl());
+    urlTemplates.append(settingsCache->getPicUrlFallback());
+
     if (card) {
         sortedSets = card->getSets();
         qSort(sortedSets.begin(), sortedSets.end(), SetDownloadPriorityComparator());
+        nextSet();
+    }
+}
+
+void PictureToLoad::populateSetUrls()
+{
+    currentSetUrls.clear();
+
+    if (card && currentSet) {
+        QString setCustomURL = card->getCustomPicURL(currentSet->getShortName());
+
+        if (!setCustomURL.isEmpty()) {
+            currentSetUrls.append(setCustomURL);
+        }
+    }
+
+    foreach (QString urlTemplate, urlTemplates) {
+        QString transformedUrl = transformUrl(urlTemplate);
+
+        if (!transformedUrl.isEmpty()) {
+            currentSetUrls.append(transformedUrl);
+        }
+    }
+
+    if (!currentSetUrls.isEmpty()) {
```

Comment:
This few lines are basically an initial call to `nextUrl()`

Reply:
good catch, I'll update this as well.